akka-stream/src/main/scala/akka/stream/scaladsl/One2OneBidiFlow.scala

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.scaladsl
+
+import akka.stream._
+
+import scala.util.control.NoStackTrace
+
+/**
+ * A generic BidiFlow which verifies that another flow produces exactly one output element per
```

Comment:
Isn't the current implemented logic "at least one output element per input element" instead of "exactly one output element"? What's the reason not to implement the "exactly one output element" rule more closely at least as far as it is dynamically possible. As far as possible because we can only associate input elements to output elements after the input and the wrapped flow are completed. So, the best dynamic approximation would be that we keep an `pendingOutput` counter that never goes below 0.

So, for a supposed identity flow an allowed sequence would be

```
input: 1
output: 1
```

but not

```
output: 1 <- fails here, because we haven't seen any input
input: 1
```

and

```
input: 1
input: 2
output: 1
input: 3
output: 2
output: 3
```

or with bugs (which this flow should prevent but is not able to in this case):

```
input: 1
input: 2
output: 1
output: 1 // oops, bug in the wrapped flow: outputs two elements for the first input, we cannot detect this
```

in which case it may fail later at

```
input: 1
input: 2
output: 1
output: 1 // oops, bug in the wrapped flow: outputs two elements for the first input
output: 2 // <- fail here
```

or

```
input: 1
input: 2
output: 1
output: 1 // oops, bug in the wrapped flow: outputs two elements for the first input
input: 3
output: 2 // oops, we mis-associated another wrong input element with the output element
output: 3 <- but at least we can fail here
```